lib/phoenix_live_view/html_tokenizer.ex

```diff
@@ -94,15 +94,20 @@ defmodule Phoenix.LiveView.HTMLTokenizer do
   end
 
   defp handle_text("<!--" <> rest, line, column, buffer, acc, state) do
+    state = update_in(state.context, &[:comment_start | &1])
     handle_comment(rest, line, column + 4, ["<!--" | buffer], acc, state)
   end
 
   defp handle_text("</" <> rest, line, column, buffer, acc, state) do
-    handle_tag_close(rest, line, column + 2, text_to_acc(buffer, acc, line, column), state)
+    text_to_acc = text_to_acc(buffer, acc, line, column, state.context)
+    state = put_in(state.context, [])
```

Comment:
We have to clean the state before putting the buffer to `acc` in order to avoid adding the `context` to tokens that are not related to `:comment`